pkg/util/labels.go

```diff
@@ -44,9 +44,10 @@ func AddObjectLabels(obj runtime.Object, labels labels.Set) error {
 			}
 		}
 
-		if err := MergeInto(metaLabels, labels, ErrorOnDifferentDstKeyValue); err != nil {
+		if err := MergeInto(metaLabels, labels, OverwriteExistingDstKey); err != nil {
```

Comment:
@smarterclayton says this was a bug, and should have been overwrite. looks like the same issue exists below for AddObjectAnnotations... fix there as well?